BTCPayServer/Services/UserService.cs

```diff
@@ -46,26 +46,64 @@ public async Task<List<ApplicationUserData>> GetUsersWithRoles()
                (userRole, role) => role.Name).ToArray()))).ToListAsync();
         }
 
-
         public static ApplicationUserData FromModel(ApplicationUser data, string?[] roles)
         {
-            return new ApplicationUserData()
+            return new ApplicationUserData
             {
                 Id = data.Id,
                 Email = data.Email,
                 EmailConfirmed = data.EmailConfirmed,
                 RequiresEmailConfirmation = data.RequiresEmailConfirmation,
+                Approved = data.Approved,
+                RequiresApproval = data.RequiresApproval,
                 Created = data.Created,
                 Roles = roles,
                 Disabled = data.LockoutEnabled && data.LockoutEnd is not null && DateTimeOffset.UtcNow < data.LockoutEnd.Value.UtcDateTime
             };
         }
 
-        private bool IsDisabled(ApplicationUser user)
+        private static bool IsEmailConfirmed(ApplicationUser user)
+        {
+            return user.EmailConfirmed || !user.RequiresEmailConfirmation;
+        }
+
+        private static bool IsApproved(ApplicationUser user)
+        {
+            return user.Approved || !user.RequiresApproval;
+        }
+
+        private static bool IsDisabled(ApplicationUser user)
         {
             return user.LockoutEnabled && user.LockoutEnd is not null &&
                    DateTimeOffset.UtcNow < user.LockoutEnd.Value.UtcDateTime;
         }
+        
+        public static bool TryCanLogin(ApplicationUser? user, out string? error)
```

Comment:
```suggestion
        public static bool TryCanLogin([NotNullWhenAttribute(true)] ApplicationUser? user, out string? error)
```

This way, if the method returns `true` you don't have to check `user` for null and can remove stuff like `user!`

Reply:
Just a note for this method, I would prefer if you used a discriminated unions pattern via records rather than sending error back as a string, this way the caller can check what error is this without having to parse the error string.

That said, don't change it now this is fine.